draft-ietf-quic-tls.md

```diff
@@ -821,6 +830,73 @@ used in QUIC can cause packet numbers to be decoded incorrectly if they are
 delayed significantly.
 
 
+# Unprotected Packets
+
+QUIC adds an integrity check to all unprotected packets.  Any packet that is not
+protected by the negotiated AEAD (see {{packet-protection}}), includes a
+integrity check.  This check does not prevent the packet from being altered, it
+exists for added resilience against data corruption and to provided added
+assurance that the sender intends to use QUIC.
+
+Unprotected packets all use the long form of the QUIC header and so will include
+a version number.  For this version of QUIC, the integrity check uses the
+128-bit FNV-1a hash (see {{fnv1a}}).  The output of this hash is appended to the
+payload of the packet.
+
+The integrity check algorithm MAY change for other versions of the protocol.
+
+
+## Integrity Check Processing
+
+An endpoint sending a packet that has a long header and a type that does not
+indicate that the packet will be protected (that is, 0-RTT Encrypted (0x05),
+1-RTT Encrypted (key phase 0) (0x06), or 1-RTT Encrypted (key phase 1) (0x07))
+first constructs the packet that it sends without the integrity check.
+
+In constructing the packet, the endpoint has to consider the extra length that
+the integrity check will add relative to the path MTU.  In particular, the
+integrity check length needs to be subtracted from available space when padding
+the initial client packet.
+
+The sender then calculates the integrity check over the entire packet, starting
+from the type field.  The output of the hash is appended to the packet.
```

Comment:
Google QUIC adds this hash right after the packet header. This is similar to how AEAD does it, where the hash appears just before the encrypted payload, and it is indeed how we generate these packets (null encrypter with an FNV hash.) Can you change this to have the hash after the packet header?